### PR TITLE
Feature/Redis Trigger

### PR DESCRIPTION
**Pull request by suneetnangia**
Added Redis trigger support in addition to existing http in the shim.
I'll add integration test support for this and also the existing "spin-outbound-redis" in next PR.

We are currently using a private version of this shim in our repo to demo wasm/containers messaging using Redis in K3d. https://github.com/suneetnangia/wasm-orchestration-with-spin

P.S. Let me know if you want to make any changes. Thanks

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -1,23 +1,33 @@
   {
-    "name": "Codespace to bootstrap k3d in a Codespace",
+    "name": "Containerd Wasm Shims",
```
Comment (Mossaka) [inline .devcontainer/devcontainer.json:2]: Could you please rebase to `main` since #116 merged.

File: containerd-shim-spin-v1/Cargo.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "containerd-shim-spin-v1"
-version = "0.8.0"
+version = "0.9.0"
```
Comment (Mossaka) [inline containerd-shim-spin-v1/Cargo.toml:3]: Unforunately I don't thnk we current allow bumping invidiual shim's verison our release pipeline is using one version for all the shims.

File: containerd-shim-spin-v1/src/main.rs [hunk 3]
```diff
@@ -183,12 +169,64 @@ impl Instance for Wasi {
...
+                                    ))))
+                                    .unwrap();
+                                    return;
+                                }
+                            };
+
+                            info!(" >>> running spin trigger");
+                            f = redis_trigger.run(spin_trigger::cli::NoArgs);
+                        }
+                        _ => todo!("Only Http and Redis triggers are currently supported."),
```
Comment (Mossaka) [inline containerd-shim-spin-v1/src/main.rs:228]: I am working on a generic trigger implementaion for the spin shim. This definitely helps!

**Review by Mossaka — APPROVED** [review]
LGTM! Made a few comments.

FYI, I am working on upgrading the spin shim to use `youki` container runtime, and in the process of doing that, I am thinking about using generic trigger from Spin so that it not only supports http and redis, but any spin custom triggers. But I do think this PR is mergable.

**Comment by suneetnangia** [comment]
All done @Mossaka. Thanks

**Review by Mossaka — APPROVED** [review]
lgtm!

Would you like to add the `spin-inbound-redis` image to the release pipeline?  @suneetnangia

**Comment by Mossaka** [comment]
I am going to merge this in and we can have follow ups resolved in future PRs.

**Comment by suneetnangia** [comment]
> lgtm!
> 
> Would you like to add the `spin-inbound-redis` image to the release pipeline? @suneetnangia

Can do, but was wondering if I should add integration tests for redis based images (both inbound and outbound (existing)) prior to release, to ensure images always work when we make changes?

**Comment by Mossaka** [comment]
@suneetnangia it's always a good idea to add more integration tests, specifically for redis based images.

**Pull request merged**